### ISSUE TITLE
Deleting product doesn't remove it from any Grouped Products

### ISSUE THREAD

**Issue by majemedia** [woocommerce/woocommerce#23985]
**Describe the bug**
When you delete (not just trash) a product that is part of a grouped product the _children meta of any grouped product that it is attached to isn't updated.

**To Reproduce**
1. Create product
2. attach to grouped product
3. save grouped product
4. view `_children` meta value of grouped product
5. Permanently delete product
6. view `_children` meta value of grouped product

**Expected behavior**
If a product that is part of a grouped product is removed it should be removed from that grouped product

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

I don't believe this is related to #1786

**Comment by vedanshujain** [comment]
Right now we do not store any reference to the grouped product inside an individual product which is being grouped, so we would have to go through all grouped products whenever an individual product is deleted. This could be very inefficient. 

As a workaround, you can use `wc_products_array_filter_visible_grouped` function to filter out all child product that can be displayed, which is also used in core do display grouped product in frontend and in admin view. 

I am marking this won't fix, but we can re-evaluate when we have more bandwidth, or if there is a very compelling reason to address this, which can not be  handled by current workaround.

**Comment by majemedia** [comment]
I'm okay with this staying closed. I'm offering this comment purely as a future implementation idea.

Because of a client need for a solution I created a plugin that essentially:

- Adds a `_parent` meta key with a serialized array of values to a child product that includes any parent product's id
- Upon creation of a grouped product each child product is looped through to update/create the meta_key
- Upon update of a grouped product the difference in child products (added or removed) is used to update the child product's status
- Upon deletion (not trashing) of a parent product each child product is gone through and has the parent product removed
- Upon deletion (not trashing) of a child product each parent product in the `_parent` array is updated to remove

I definitely agree that the update of an existing site utilizing grouped products may be time consuming but could be realistically done in version update.

The client I built this solution for has 90 grouped products that contain a mixture of children across 600 skus with one grouped product containing 100 children.

The update script as written took a total of 8 seconds to execute using the proper WC CRUD methods.
